app/controllers/admin/users_controller.rb

```diff
@@ -403,8 +403,12 @@ def destroy
   end
 
   def destroy_bulk
+    # capture service_params outside the hijack block to avoid thread safety
+    # issues
+    service_arg = service_params.merge(options: { actor_ip: current_user.ip_address })
```

Comment:
Why does `actor_ip` need to be split out here? We already have `guardian.user` in the service, which is `current_user`. Capturing outside the block makes sense though, @Flink might just want to be aware of this generally if not already.

Reply:
Not sure to follow why we have to do this? As @martin-brennan says, we already have a `guardian` inside the service. And it’s already used by the `can_delete_users` policy. So why would it create a threading issue to access the current user’s IP? (I may be missing something, that’s why I ask)

Reply:
@Flink the thread safety issue comes from the `hijack` call not the service. `hijack` processes the request in a background thread and frees up the main thread to serve other requests. If you invoke a controller method, e.g. the `current_user` method, in a `hijack` block, there's a possibility that the method could return the user of another request to a completely different endpoint rather than the user that called the `hijack`ed endpoint. We've seen this issue several times (can't find examples at the moment) and our "fix" has been to call any controller methods outside the `hijack` block and stash their values in local variables which we then reference inside `hijack`.